### PR TITLE
feat: Deprecated Table in Main

### PR DESCRIPTION
**Pull request by thunguyen19**
## Summary

Deprecate `Table` in main.

Resolves: #2249 

## Release Category
Components

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)

## Where Should the Reviewer Start?

`modules/react/table/lib/Table.tsx`
`modules/react/table/lib/TableRow.tsx`
`modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx`
`/.storybook/routes.js/` is already updated with route to Table in Preview https://github.com/Workday/canvas-kit/blob/prerelease/major/.storybook/routes.js#L15

### CODE REVIEW

File: modules/react/table/lib/Table.tsx [hunk 1]
```diff
@@ -16,6 +16,11 @@ const TableComponent = styled('table')(type.levels.subtext.large, {
   },
 });
 
+/**
+ * ### ⚠️ Table has been deprecated and will be removed in v10 ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/Table.tsx:20]:
```suggestion
 * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```

File: modules/react/table/lib/Table.tsx [hunk 2]
```diff
@@ -16,6 +16,11 @@ const TableComponent = styled('table')(type.levels.subtext.large, {
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in preview
```
Comment (mannycarrera4) [inline modules/react/table/lib/Table.tsx:21]:
```suggestion
 * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```

File: modules/react/table/lib/TableRow.tsx [hunk 3]
```diff
@@ -5,6 +5,11 @@ import {colors, space, spaceNumbers, statusColors} from '@workday/canvas-kit-rea
 import {createComponent, StyledType} from '@workday/canvas-kit-react/common';
 import {borderColor, borderWidth, cellBorder} from './Table';
 
+/**
+ * ### ⚠️ Table has been deprecated and will be removed in v10 ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:9]:
```suggestion
 * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```

File: modules/react/table/lib/TableRow.tsx [hunk 4]
```diff
@@ -5,6 +5,11 @@ import {colors, space, spaceNumbers, statusColors} from '@workday/canvas-kit-rea
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in preview
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:10]:
```suggestion
 * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```

File: modules/react/table/lib/TableRow.tsx [hunk 5]
```diff
@@ -14,6 +19,11 @@ export enum TableRowState {
   Selected,
 }
 
+/**
+ * ### ⚠️ Table has been deprecated and will be removed in v10 ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:23]:
```suggestion
 * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```

File: modules/react/table/lib/TableRow.tsx [hunk 6]
```diff
@@ -14,6 +19,11 @@ export enum TableRowState {
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in preview
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:24]:
```suggestion
 * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```

File: modules/react/table/lib/TableRow.tsx [hunk 7]
```diff
@@ -192,6 +202,11 @@ const StyledTableRow = styled('tr')<TableRowProps>(
   }
 );
 
+/**
+ * ### ⚠️ Table has been deprecated and will be removed in v10 ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:206]:
```suggestion
 * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```

File: modules/react/table/lib/TableRow.tsx [hunk 8]
```diff
@@ -192,6 +202,11 @@ const StyledTableRow = styled('tr')<TableRowProps>(
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in preview
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:207]:
```suggestion
 * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```

**Comment by mannycarrera4** [comment]
We should add this to our comms in `10.0-UPGRADE-GUIDE.mdx`. You can reference our messaging to other things that are being deprecated

File: modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx [hunk 9]
```diff
@@ -13,6 +13,8 @@ any questions.
 	- [useBanner](#useBanner)
 - [Deprecations](#deprecations)
   - [InputIconContainer](#input-icon-container)
+  - [Table](#table)
+  - [TableRow](#table-row)
```
Comment (mannycarrera4) [inline modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx:17]: You can probably just reference `Table` here and down below mention that all exported members or components are being deprecated

Comment (thunguyen19) [reply modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx]: Updated. Could you re-check?

File: modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx [hunk 10]
```diff
@@ -70,6 +71,10 @@ and browsers can compute the name of the Banner from the text given inside.
 We've deprecated `InputIconContainer` from [Main](#main) because it doesn't handle
 bidirectionality or icons at the start of an input. Please consider using [`InputGroup`](https://workday.github.io/canvas-kit/?path=/story/components-inputs-text-input--icons) instead.
 
+### Table
+
+We've deprecated `Table`, `TableRow` components and all of their exported members. Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) instead as it is more flexible and modern.
```
Comment (mannycarrera4) [inline modules/docs/mdx/10.0-UPGRADE_GUIDE.mdx:76]:
```suggestion
We've deprecated `Table` and `TableRow` as well as all of their exported members. Please consider using [`Table in Preview`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) instead.
```

File: modules/react/table/lib/Table.tsx [hunk 11]
```diff
@@ -16,6 +16,11 @@ const TableComponent = styled('table')(type.levels.subtext.large, {
   },
 });
 
+/**
+ * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```
Comment (mannycarrera4) [inline modules/react/table/lib/Table.tsx:21]:
```suggestion
 * Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```

File: modules/react/table/lib/Table.tsx [hunk 12]
```diff
@@ -16,6 +16,11 @@ const TableComponent = styled('table')(type.levels.subtext.large, {
   },
 });
 
+/**
+ * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/Table.tsx:20]:
```suggestion
 * @deprecated ⚠️ `Table` has been deprecated and will be removed in a future major version.
```

File: modules/react/table/lib/Table.tsx [hunk 13]
```diff
@@ -16,6 +16,11 @@ const TableComponent = styled('table')(type.levels.subtext.large, {
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
+ * @deprecated
```
Comment (mannycarrera4) [inline modules/react/table/lib/Table.tsx:22]:
```suggestion
```

File: modules/react/table/lib/TableRow.tsx [hunk 14]
```diff
@@ -14,6 +19,11 @@ export enum TableRowState {
   Selected,
 }
 
+/**
+ * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:23]:
```suggestion
 * @deprecated ⚠️ `Table` has been deprecated and will be removed in a future major version.
```

File: modules/react/table/lib/TableRow.tsx [hunk 15]
```diff
@@ -14,6 +19,11 @@ export enum TableRowState {
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:24]:
```suggestion
 * Please consider using [`Table in Preview`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) instead.
```

File: modules/react/table/lib/TableRow.tsx [hunk 16]
```diff
@@ -192,6 +202,11 @@ const StyledTableRow = styled('tr')<TableRowProps>(
   }
 );
 
+/**
+ * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
+ * @deprecated
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:208]:
```suggestion
```

File: modules/react/table/lib/TableRow.tsx [hunk 17]
```diff
@@ -192,6 +202,11 @@ const StyledTableRow = styled('tr')<TableRowProps>(
   }
 );
 
+/**
+ * ### ⚠️ `Table` has been deprecated and will be removed in a future major version. ⚠️
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:206]:
```suggestion
 * @deprecated ⚠️ `Table` has been deprecated and will be removed in a future major version.
```

File: modules/react/table/lib/TableRow.tsx [hunk 18]
```diff
@@ -192,6 +202,11 @@ const StyledTableRow = styled('tr')<TableRowProps>(
...
+ * - Please consider using [`Table`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) in Preview.
```
Comment (mannycarrera4) [inline modules/react/table/lib/TableRow.tsx:207]:
```suggestion
 * Please consider using [`Table in Preview`](https://workday.github.io/canvas-kit/?path=/docs/preview-table--basic) instead.
```

**Review by mannycarrera4 — APPROVED** [review]

**Pull request merged**